api/src/resources/dataset.rs

```diff
@@ -244,6 +244,9 @@ pub struct NewDataset<'request> {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub entity_defs: Option<&'request [NewEntityDef]>,
 
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub general_fields: Option<&'request [NewGeneralFieldDef]>,
+
```

Comment:
Could you also add it to `Dataset`? such that we can easily roundtrip them :pray: